app/controllers/journeys/expenses/FinancialExpensesController.scala

```diff
@@ -42,28 +48,50 @@ class FinancialExpensesController @Inject() (override val messagesApi: MessagesA
     extends FrontendBaseController
     with I18nSupport {
 
-  val form = formProvider()
+  val businessId = "SJPR05893938418"
+  val taxYear    = LocalDate.now.getYear
 
-  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData) { implicit request =>
-    val preparedForm = request.userAnswers.getOrElse(UserAnswers(request.userId)).get(FinancialExpensesPage) match {
-      case None        => form
-      case Some(value) => form.fill(value)
-    }
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData) async { implicit request =>
+    selfEmploymentService.getAccountingType(request.user.nino, businessId, request.user.mtditid) map {
+      case Left(_) => Redirect(JourneyRecoveryController.onPageLoad())
+      case Right(accountingType) =>
+        val preparedForm = request.userAnswers.getOrElse(UserAnswers(request.userId)).get(FinancialExpensesPage, Some(businessId)) match {
+          case None        => formProvider(userType(request.user.isAgent))
+          case Some(value) => formProvider(userType(request.user.isAgent)).fill(value)
+        }
+        val legendContentString = buildLegendHeadingWithHintString(
```

Comment:
If both `legendContentString` used in `onSubmit` and here are going to always be the same, it could be worth pulling them out into a method to reduce duplication.

Also, we don't need the suffix `..String` on the end, the IDE tells us this.

Thirdly (if this is the case - ignore part 1), should the controller be doing this? Or is it more of the views responsibility to be doing this than the controller?

Reply:
Will remove string and separate it from the methods, thanks. I found this was the only way to pass both parts of the content into the the view's LegendViewModel. Had to do the same thing for the task list.